wordpress-mobile/WordPress-Android
# Feature/quick start notification

Part of #7707 

**Do not merge this PR untill [Flux-C counterpart](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/878) is merged and reference to correct commit are updated.**

This PR adds a Quick Start completion notification.
Notification should not be fired when you skip QuickStart.

In case, when Quick Start was completed offline, or notification request somehow did not reached the server, we will try to request notification when opening app again, changing site or navigating to main activity again.

This PR also adds couple of minor changes to the code:
- We are using separate flag to indicate that quick start was completed (instead of `isTaskShown`)
- When individual quick start task prompt was shown maximum number of times it will not start apearing again after you complete a task naturally (when not prompted or not by selecting it from Quick Start list)

To test:

1 - Complete a Quick Start normally.
2 - Make sure you get a notification.

1 - Complete a Quick Start while offline.
2 - Make sure you get a notification after you navigate back to main activity with connection to internet.

1 - Complete a Quick Start while offline.
2 - Switch to some other site.
3 - Connect to internet
4 - Switch back to original site with completed quick start.
5 - Make sure you get a notification.
